core/src/main/java/org/pentaho/di/core/database/BaseDatabaseMeta.java

```diff
@@ -174,6 +175,15 @@ public abstract class BaseDatabaseMeta implements Cloneable, DatabaseInterfaceEx
   public static final String AUTOINCREMENT_SQL_FOR_BATCH_ID = "AUTOINCREMENT_SQL_FOR_BATCH_ID";
   public static final String NAMED_CLUSTER_ID = "NAMED_CLUSTER_ID";
 
+  public static final List<String> ATTRIBUTE_KEYLIST = Arrays.asList( ATTRIBUTE_PORT_NUMBER,
```

Comment:
Suggest to use Map/Enum.
If going ahead with map then map can be initialized. and values can be put from static block.